src/data/equiv/transfer_instance.lean

```diff
@@ -75,105 +77,94 @@ the one obtained by transporting a multiplicative structure on `β` back along `
-let mul := e.has_mul in
-by resetI; apply e.injective.comm_semigroup _; intros; exact e.apply_symm_apply _
+@injective.comm_semigroup α β e.has_mul _ e e.injective $
+  λ x y, e.apply_symm_apply _
 
 /-- Transfer `mul_zero_class` across an `equiv` -/
 protected def mul_zero_class [mul_zero_class β] : mul_zero_class α :=
-let zero := e.has_zero, mul := e.has_mul in
-by resetI; apply e.injective.mul_zero_class _; intros; exact e.apply_symm_apply _
+@injective.mul_zero_class _ _ _ e.has_mul e.has_zero e e.injective
+  (e.apply_symm_apply _) (λ x y, e.apply_symm_apply _)
 
 /-- Transfer `mul_one_class` across an `equiv` -/
 @[to_additive "Transfer `add_zero_class` across an `equiv`"]
 protected def mul_one_class [mul_one_class β] : mul_one_class α :=
-let one := e.has_one, mul := e.has_mul in
-by resetI; apply e.injective.mul_one_class _; intros; exact e.apply_symm_apply _
+@injective.mul_one_class _ _ e.has_mul e.has_one _ e e.injective
+  (e.apply_symm_apply _) (λ x y, e.apply_symm_apply _)
 
 /-- Transfer `mul_zero_one_class` across an `equiv` -/
 protected def mul_zero_one_class [mul_zero_one_class β] : mul_zero_one_class α :=
-let zero := e.has_zero, one := e.has_one,mul := e.has_mul in
-by resetI; apply e.injective.mul_zero_one_class _; intros; exact e.apply_symm_apply _
+@injective.mul_zero_one_class _ _ _ e.has_mul e.has_zero e.has_one e e.injective
+  (e.apply_symm_apply _) (e.apply_symm_apply _) (λ x y, e.apply_symm_apply _)
 
 /-- Transfer `monoid` across an `equiv` -/
 @[to_additive "Transfer `add_monoid` across an `equiv`"]
 protected def monoid [monoid β] : monoid α :=
-let one := e.has_one, mul := e.has_mul in
-by resetI; apply e.injective.monoid _; intros; exact e.apply_symm_apply _
+{ npow := λ n x, e.symm (e x ^ n),
+  npow_zero' := λ x, congr_arg e.symm (pow_zero _),
+  npow_succ' := λ n x, congr_arg e.symm $ by rw [pow_succ, e.apply_symm_apply],
+  .. @injective.monoid _ _ e.has_mul e.has_one _ e e.injective
+    (e.apply_symm_apply _) (λ x y, e.apply_symm_apply _) }  
 
 /-- Transfer `comm_monoid` across an `equiv` -/
 @[to_additive "Transfer `add_comm_monoid` across an `equiv`"]
 protected def comm_monoid [comm_monoid β] : comm_monoid α :=
-let one := e.has_one, mul := e.has_mul in
-by resetI; apply e.injective.comm_monoid _; intros; exact e.apply_symm_apply _
+{ .. e.monoid, .. e.comm_semigroup }
+
+/-- Transfer `div_inv_monoid` across an `equiv` -/
+@[to_additive "Transfer `sub_neg_monoid` across an `equiv`"]
+protected def div_inv_monoid [div_inv_monoid β] : div_inv_monoid α :=
+{ zpow := λ n x, e.symm (e x ^ n),
+  zpow_zero' := λ x, congr_arg e.symm (zpow_zero _),
+  zpow_succ' := λ n a, congr_arg e.symm $ (div_inv_monoid.zpow_succ' _ _).trans $
+    congr_arg _ (e.apply_symm_apply _).symm,
+  zpow_neg' := λ n x, congr_arg e.symm $ (div_inv_monoid.zpow_neg' _ _).trans $
+    congr_arg _ (e.apply_symm_apply _).symm,
+  .. e.monoid,
+  .. @injective.div_inv_monoid α β e.has_mul e.has_one e.has_inv e.has_div _ e e.injective
```

Comment:
This can now use `div_inv_monoid_pow`, right?

Reply:
Yes but I have no time for this PR today, then I'll have to grade a large exam. If you want it to be merged soon, then feel free to adopt it.

Reply:
I'm not in any rush, I was just skimming over stale PRs